AEPLifecycle/Sources/LifecycleV2/LifecycleV2DataStoreCache.swift

```diff
@@ -0,0 +1,76 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPServices
+import Foundation
+
+/// Lifecycle DataStore Cache layer for persisting the date values required for Lifecycle session computation in XDM,
+/// including close date to be used for app close time approximation, start and pause dates.
+class LifecycleV2DataStoreCache {
+    private var closeDate: Date?
+    private var lastClosePersistedDate: Date?
+    private let dataStore: NamedCollectionDataStore
+
+    init(dataStore: NamedCollectionDataStore) {
+        self.dataStore = dataStore
+        if let persistedCloseDate = dataStore.getObject(key: LifecycleV2Constants.DataStoreKeys.APP_CLOSE_DATE) as Date? {
+            self.lastClosePersistedDate = Date(timeIntervalSince1970: persistedCloseDate.timeIntervalSince1970)
```

Comment:
this should not be initialized here, it will be initialized with first event seen in current session.

Reply:
If we remove this, it would not work for further launches right? We wont have this value when app is closed and reopened since its in persistence but the variable might not be holding the value from the first launch.

Reply:
`lastClosePersistedDate` was added to store the latest updates in current session, as an optimization so we don’t update on each event, so it is ok to be 0 for the first event processed in current session.
Your compare in L34 will be currentDate - 0 > 2 ✔️

Reply:
Yes, got it. Have made the change.